Sources/ReportingService.swift

```diff
@@ -118,58 +99,56 @@ class ReportingService {
       type: .step
     )
     
-    try httpClient.callEndPoint(endPoint) { (result: Item) in
-      self.testID = result.id
-      self.semaphore.signal()
-    }
-    _ = semaphore.wait(timeout: .now() + timeOutForRequestExpectation)
-  }
-  
-  func reportError(message: String) throws {
-    let endPoint = PostLogEndPoint(itemID: testID, level: "error", message: message)
-    try httpClient.callEndPoint(endPoint) { (result: Item) in
-      self.semaphore.signal()
-    }
-    _ = semaphore.wait(timeout: .now() + timeOutForRequestExpectation)
+    let response: Result<Item, Error> = self.httpClient.synchronousCallEndPoint(endPoint)
+    switch response {
+      case .success(let result):
+        self.testID = result.id
+      case .failure(let error):
+        print(error)
+     }
+    
+     self.fileService!.createLogFile(withName: extractTestName(from: test))
+   }
+    
+  func reportLog(level: String, message: String) throws {
+    let endPoint = PostLogEndPoint(itemID: testID, level: level, message: message)
+    
+    let _: Result<Item, Error> = self.httpClient.synchronousCallEndPoint(endPoint)
   }
-  
+    
   func finishTest(_ test: XCTestCase) throws {
     let testStatus = test.testRun!.hasSucceeded ? TestStatus.passed : TestStatus.failed
     if testStatus == .failed {
       testSuiteStatus = .failed
       launchStatus = .failed
     }
     
+    try? reportLog(level: "info", message: fileService!.readLogFile(fileName: extractTestName(from: test)))
+    fileService!.deleteLogFile(withName: extractTestName(from: test))
+    
     let endPoint = FinishItemEndPoint(itemID: testID, status: testStatus)
     
-    try httpClient.callEndPoint(endPoint) { (result: Finish) in
-      self.semaphore.signal()
-    }
-    _ = semaphore.wait(timeout: .now() + timeOutForRequestExpectation)
+    let _: Result<Finish, Error> = self.httpClient.synchronousCallEndPoint(endPoint)
```

Comment:
One more comment here. You can mark `synchronousCallEndPoint` with `@discardableResult` which will allow you to drop the `let _: Result<Finish, Error> =` at the call sites. It won't warn anymore if you don't use the result from the function.

Reply:
Unfortunately, I can’t implement @discardableResult for the func synchronousCallEndPoint.
The function return parametrized value. Therefore, in case I add func synchronousCallEndPoint and make a simple call self.httpClient.synchronousCallEndPoint(endPoint) I receive an error:
Generic parameter ‘T’ could not be inferred